scripts/solve_network.py

```diff
@@ -346,6 +346,116 @@ def add_battery_constraints(n):
     define_constraints(n, lhs, "=", 0, "Link", "charger_ratio")
 
 
+def add_RES_constraints(n, res_share):
+    lgrouper = n.loads.bus.map(n.buses.country)
+    ggrouper = n.generators.bus.map(n.buses.country)
+    sgrouper = n.storage_units.bus.map(n.buses.country)
+    cgrouper = n.links.bus0.map(n.buses.country)
+
+    load = (
+        n.snapshot_weightings.generators
+        @ n.loads_t.p_set.groupby(lgrouper, axis=1).sum()
+    )
+
+    rhs = res_share * load
+
+    res_techs = [
+        "solar",
+        "onwind",
+        "offwind-dc",
+        "offwind-ac",
+        "battery",
+        "hydro",
+        "ror",
+    ]
+    charger = ["H2 electrolysis", "battery charger"]
+    discharger = ["H2 fuel cell", "battery discharger"]
+
+    gens_i = n.generators.query("carrier in @res_techs").index
+    stores_i = n.storage_units.query("carrier in @res_techs").index
+    charger_i = n.links.query("carrier in @charger").index
+    discharger_i = n.links.query("carrier in @discharger").index
+
+    # Generators
+    lhs_gen = (
+        linexpr(
+            (n.snapshot_weightings.generators, get_var(n, "Generator", "p")[gens_i].T)
+        )
+        .T.groupby(ggrouper, axis=1)
+        .apply(join_exprs)
+    )
+
+    # StorageUnits
+    lhs_dispatch = (
+        (
+            linexpr(
+                (
+                    n.snapshot_weightings.stores,
+                    get_var(n, "StorageUnit", "p_dispatch")[stores_i].T,
+                )
+            )
+            .T.groupby(sgrouper, axis=1)
+            .apply(join_exprs)
+        )
+        .reindex(lhs_gen.index)
+        .fillna("")
+    )
+
+    lhs_store = (
+        (
+            linexpr(
+                (
+                    -n.snapshot_weightings.stores,
+                    get_var(n, "StorageUnit", "p_store")[stores_i].T,
+                )
+            )
+            .T.groupby(sgrouper, axis=1)
+            .apply(join_exprs)
+        )
+        .reindex(lhs_gen.index)
+        .fillna("")
+    )
+
+    # Stores (or their resp. Link components)
+    # Note that the variables "p0" and "p1" currently do not exist.
+    # Thus, p0 and p1 must be derived from "p" (which exists), taking into account the link efficiency.
+    lhs_charge = (
+        (
+            linexpr(
+                (
+                    -n.links.loc[charger_i].efficiency,
+                    get_var(n, "Link", "p")[charger_i],
+                )
+            )
+            .groupby(cgrouper, axis=1)
+            .apply(join_exprs)
+        )
+        .reindex(lhs_gen.index)
+        .fillna("")
+    )
+
+    lhs_discharge = (
+        (
+            linexpr(
+                (
+                    n.links.loc[discharger_i].efficiency,
```

Comment:
```suggestion
                    1./n.links.loc[discharger_i].efficiency,
```

I think this would be correct, for the reviewer. Maybe they can double-check. I have an evaluation ipynb prebared, it is available to both @yerbol-akhmetov  and @pz-max

Reply:
Agree, it should be 1/eta

Reply:
As discussed above, I think here the "n.links.loc[discharger_i].efficiency" is ok